lib/ansible/modules/utilities/logic/validate_arg_spec.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright 2018 Red Hat
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['stableinterface'],
+                    'supported_by': 'core'}
+
+
+DOCUMENTATION = '''
+---
+module: validate_arg_spec
+short_description: Validate Arg Specs
+description:
+     - This module validate args specs
+version_added: "2.8"
+options:
+  argument_spec:
+    description:
+        - A dictionary like AnsibleModule argument_spec
+    required: true
+  provided_arguments:
```

Comment:
we should not need to provide arguments, the 'current variables' are what we want to check against

see example on use with `role_params` below for specifics